### ISSUE TITLE
[Feature] resource_cluster data source

### ISSUE THREAD

**Issue by scott-the-programmer** [scott-the-programmer/terraform-provider-minikube#119]
To be close to feature complete, we should implement the ability to reference an existing minikube cluster as a [data source](https://developer.hashicorp.com/terraform/language/data-sources)

The goal is to allow users to reference previously created clusters (either created via the cli, or through a separate terraform stack) by specifying something like

```terraform
data "minikube_cluster" "some_cluster" {
    cluster_name = "some_existing_cluster_name"
}

provider "kubernetes" {
  host = minikube_cluster.some_cluster.host

  client_certificate     = data.minikube_cluster.some_cluster.client_certificate
  client_key             = data.minikube_cluster.some_cluster.client_key
  cluster_ca_certificate = data.minikube_cluster.some_cluster.cluster_ca_certificate
}
```

**Comment by caerulescens** [comment]
Awesome feature, can't wait.

**Comment by caerulescens** [comment]
I could use this functionality; may I implement it?

**Comment by scott-the-programmer** [comment]
> I could use this functionality; may I implement it?

Yup, that would be awesome! I haven't had much time lately, so any help would be appreciated. 

I suspect that the implementation would be fairly easy, i.e. pulling the existing config via a cluster name - but i'm not entirely sure if there's a convenient call in minikube to grab this

**Comment by caerulescens** [comment]
Great, and that's what I was expecting too; nothing too difficult.

**Comment by caerulescens** [comment]
I'm actually going to need this functionality at work, so I'll address this.
